File: backend/app/models/submission.py
from sqlalchemy import Column, Integer, String, Text, DateTime, JSON

from app.core.database import Base


class Submission(Base):
    __tablename__ = "submissions"

    id = Column(Integer, primary_key=True, index=True)
    task_id = Column(Integer, nullable=True)
    dataset_item_id = Column(Integer, nullable=True)
    labeler_id = Column(Integer, nullable=True)
    data = Column(JSON, nullable=False)
    status = Column(String(32), nullable=False)
    rejected_reason = Column(Text, nullable=True)
    revision_no = Column(Integer, nullable=True)
    created_at = Column(DateTime, nullable=True)
    updated_at = Column(DateTime, nullable=True)

File: frontend/src/types/submission.ts
export interface Submission {
  id: number;
  task_id: number;
  dataset_item_id: number;
  labeler_id: number;
  status: string;
  revision_no?: number;
  data?: Record<string, unknown>;
  label_data?: any;
  submission_data?: Record<string, unknown>;
  created_at?: string;
  updated_at?: string;
}